lib/helpers.js

```diff
@@ -219,9 +219,9 @@ export function getYoungestInputTx(plasma, tx) {
 export function getProof(plasma, tx, periodOpts = {}) {
   return plasma.getPeriodByBlockHeight(tx.blockNumber)
     .then(periodData => {
-      if (periodData && periodData.length) {
+      if (periodData) {
         Object.assign(periodOpts, {
-          validatorData: periodData[0],
+          validatorData: periodData,
```

Comment:
```suggestion
          validatorData: periodData,
          excludePrevHashFromProof: true
```
I'm not sure, but seems you missed this, because I'm receiving wrong proofs without it.